pkg/cloudprovider/aws/instance.go

```diff
@@ -29,39 +29,67 @@ import (
 	v1 "k8s.io/api/core/v1"
 )
 
+const (
+	// maxInstanceTypes defines the number of instance type options to pass to fleet
+	maxInstanceTypes = 10
+)
+
 type InstanceProvider struct {
 	ec2api ec2iface.EC2API
 	vpc    *VPCProvider
 }
 
 // Create an instance given the constraints.
+// instanceTypeOptions should be sorted by priority for spot capacity type.
+// If spot is not used, the instanceTypeOptions are not required to be sorted
+// because we are using ec2 fleet's lowest-price OD allocation strategy
 func (p *InstanceProvider) Create(ctx context.Context,
 	launchTemplate *ec2.LaunchTemplate,
 	instanceTypeOptions []*packing.Instance,
 	zonalSubnetOptions map[string][]*ec2.Subnet,
+	capacityType string,
 ) (*string, error) {
-	// 1. Construct override options.
+	// 1. Trim the instanceTypeOptions so that the fleet request doesn't get too large
```

Comment:
Could you add why it can't get too large?

Reply:
added